From patchwork Tue Sep  6 19:48:56 2022
Content-Type: text/plain; charset="utf-8"
MIME-Version: 1.0
Content-Transfer-Encoding: 7bit
X-Patchwork-Submitter: Liam Howlett <Liam.Howlett@Oracle.com>
X-Patchwork-Id: 12968091
Return-Path: <owner-linux-mm@kvack.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from kanga.kvack.org (kanga.kvack.org [205.233.56.17])
	by smtp.lore.kernel.org (Postfix) with ESMTP id 52E7FC6FA86
	for <linux-mm@archiver.kernel.org>; Tue,  6 Sep 2022 19:50:02 +0000 (UTC)
Received: by kanga.kvack.org (Postfix)
	id 8AEDC80010; Tue,  6 Sep 2022 15:49:51 -0400 (EDT)
Received: by kanga.kvack.org (Postfix, from userid 40)
	id 71C8980015; Tue,  6 Sep 2022 15:49:51 -0400 (EDT)
X-Delivered-To: int-list-linux-mm@kvack.org
Received: by kanga.kvack.org (Postfix, from userid 63042)
	id 4A7E580012; Tue,  6 Sep 2022 15:49:51 -0400 (EDT)
X-Delivered-To: linux-mm@kvack.org
Received: from relay.hostedemail.com (smtprelay0017.hostedemail.com
 [216.40.44.17])
	by kanga.kvack.org (Postfix) with ESMTP id 05B6380010
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 15:49:51 -0400 (EDT)
Received: from smtpin17.hostedemail.com (a10.router.float.18 [10.200.18.1])
	by unirelay07.hostedemail.com (Postfix) with ESMTP id CD0DC1608A0
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:50 +0000 (UTC)
X-FDA: 79882700940.17.545305D
Received: from mx0a-00069f02.pphosted.com (mx0a-00069f02.pphosted.com
 [205.220.165.32])
	by imf20.hostedemail.com (Postfix) with ESMTP id 4A6C11C0091
	for <linux-mm@kvack.org>; Tue,  6 Sep 2022 19:49:50 +0000 (UTC)
Received: from pps.filterd (m0246617.ppops.net [127.0.0.1])
	by mx0b-00069f02.pphosted.com (8.17.1.5/8.17.1.5) with ESMTP id
 286IdYVR029915;
	Tue, 6 Sep 2022 19:49:43 GMT
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed; d=oracle.com;
 h=from : to : cc :
 subject : date : message-id : references : in-reply-to : content-type :
 content-transfer-encoding : mime-version; s=corp-2022-7-12;
 bh=GJ+7dVHdLllKMNXTH8C4IlcGRwqY5aNlvrMEbXEver4=;
 b=ObTpfPtUt66X8crW7FdP2tWi6I58zZd4UIPkOufVBji7fH8GK0xHhk9bwHgs8tc/P9iC
 LWCV+oL6qX08wtOcU3XPvGeHNIfD8CmLUY0OgePchwizjxG2ZSb+9oMr7cvOyirPK/dw
 8/wiaNxzIUw8VNAj74rwSGiFyIWqZmSSvtx98wD23IMf9TaL9lT5dUdmwN6vtTDnTiWb
 9HPgmSWYdRB/WlhkR30S0GEt7yxyGIq1H6DL0eftLxLHif/5xjABtJOh/o7YqMa3nvio
 IDRGGliumFgQSfHT+Zp5xxMj6I9i4fAI+6Ehk87OycfMx/V1Of5E1kDlQOwKLR69PNVO OA==
Received: from iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com
 (iadpaimrmta03.appoci.oracle.com [130.35.103.27])
	by mx0b-00069f02.pphosted.com (PPS) with ESMTPS id 3jbyftpvd3-1
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:43 +0000
Received: from pps.filterd
 (iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com [127.0.0.1])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (8.17.1.5/8.17.1.5)
 with ESMTP id 286Je5WT039711;
	Tue, 6 Sep 2022 19:49:42 GMT
Received: from nam10-dm6-obe.outbound.protection.outlook.com
 (mail-dm6nam10lp2108.outbound.protection.outlook.com [104.47.58.108])
	by iadpaimrmta03.imrmtpd1.prodappiadaev1.oraclevcn.com (PPS) with ESMTPS id
 3jbwc9m7j9-6
	(version=TLSv1.2 cipher=ECDHE-RSA-AES256-GCM-SHA384 bits=256 verify=OK);
	Tue, 06 Sep 2022 19:49:42 +0000
ARC-Seal: i=1; a=rsa-sha256; s=arcselector9901; d=microsoft.com; cv=none;
 b=P8ceZam8PMr5ccs2q7NjztM8X8d1Bn5PATGm1ajOgvLukFV4U/WQfbw3SMpvOKdVlCojqGuBEcPrst43K/MAuIkZI3ACZOmvGG39gCCZfa9pe0Q8LG8dLsC1mlZwQkxGsFULYV6ggwmNMpl53WWaQn1gHTGtIFWvkqjojPxdNm5rEmcLBO9aAgz4vsAj8LJWw6aIX7D+ibbUemfA0cp2a81OMo8PRnGPNQla9hW1uBg0EBlf7P/XNW+2JF+hmYcfvdRHkQmZSs93oJe+bWuoZkPGE0rG5fTV/pdxf0nP84Lq3w5+O6cMG+sN/CuCiEUBjjJSUjslIPY3NGNWqxsj9w==
ARC-Message-Signature: i=1; a=rsa-sha256; c=relaxed/relaxed; d=microsoft.com;
 s=arcselector9901;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-AntiSpam-MessageData-ChunkCount:X-MS-Exchange-AntiSpam-MessageData-0:X-MS-Exchange-AntiSpam-MessageData-1;
 bh=GJ+7dVHdLllKMNXTH8C4IlcGRwqY5aNlvrMEbXEver4=;
 b=cr6rdfsDM4Ec256jYQ+vM7ZnM0S3mRk8VocN1rVLCd37nv0Yjz9sudE/wH8MG+F8PZm3thJCMd5rH56oXlBrgmmbklqs5VcXPhBjdCl9tyla1p4L7vlBKK7/fe1T3yqSa03RiBRCVZGnwMrezkOfDIh7q60DqLFPEEo5qJ7RUBg/ONTIExYte/Z6n6LcIpWHBemStRnQZfHrrqfnzqJDV6MYf5S+SPX7ANj2Cv+vIOPA5eiWU8c7zDX3X5oayRFCex9OnWv2isihriUkw2LfvV1sFi2FIE2i83eUClFd0449yBup8HJVSF6jDHA+wHyOVicQ1am9O5v++9XI/5lI3Q==
ARC-Authentication-Results: i=1; mx.microsoft.com 1; spf=pass
 smtp.mailfrom=oracle.com; dmarc=pass action=none header.from=oracle.com;
 dkim=pass header.d=oracle.com; arc=none
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
 d=oracle.onmicrosoft.com; s=selector2-oracle-onmicrosoft-com;
 h=From:Date:Subject:Message-ID:Content-Type:MIME-Version:X-MS-Exchange-SenderADCheck;
 bh=GJ+7dVHdLllKMNXTH8C4IlcGRwqY5aNlvrMEbXEver4=;
 b=HOJ2NkfYHZaosyPB6ArgIv7GxPXS4qWYZgT9yd/ia3sxy+4ZrKofqxltKsKC3CNRfQ852/4kfgFyo+KT2lesGIZ+Vfa9plzYOmcGdY/fKIfPDNcu6GDlKrt0HTBYNN43cq0GobLokyc28oZubNcuEEvuXh01USwXEG/04TQ/U9U=
Received: from SN6PR10MB3022.namprd10.prod.outlook.com (2603:10b6:805:d8::25)
 by SJ0PR10MB5598.namprd10.prod.outlook.com (2603:10b6:a03:3d9::18) with
 Microsoft SMTP Server (version=TLS1_2,
 cipher=TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384) id 15.20.5588.10; Tue, 6 Sep
 2022 19:49:39 +0000
Received: from SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223]) by SN6PR10MB3022.namprd10.prod.outlook.com
 ([fe80::a420:3107:436d:d223%5]) with mapi id 15.20.5588.018; Tue, 6 Sep 2022
 19:49:39 +0000
From: Liam Howlett <liam.howlett@oracle.com>
To: "maple-tree@lists.infradead.org" <maple-tree@lists.infradead.org>,
        "linux-mm@kvack.org" <linux-mm@kvack.org>,
        "linux-kernel@vger.kernel.org"
	<linux-kernel@vger.kernel.org>,
        Andrew Morton <akpm@linux-foundation.org>
CC: "Matthew Wilcox (Oracle)" <willy@infradead.org>,
        Liam Howlett
	<liam.howlett@oracle.com>,
        Davidlohr Bueso <dave@stgolabs.net>
Subject: [PATCH v14 39/70] um: remove vma linked list walk
Thread-Topic: [PATCH v14 39/70] um: remove vma linked list walk
Thread-Index: AQHYwimzdULrjjO7wUaT0xKVMKskAA==
Date: Tue, 6 Sep 2022 19:48:56 +0000
Message-ID: <20220906194824.2110408-40-Liam.Howlett@oracle.com>
References: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
In-Reply-To: <20220906194824.2110408-1-Liam.Howlett@oracle.com>
Accept-Language: en-US
Content-Language: en-US
X-MS-Has-Attach: 
X-MS-TNEF-Correlator: 
x-mailer: git-send-email 2.35.1
x-ms-publictraffictype: Email
x-ms-office365-filtering-correlation-id: af2b36f8-81cf-412b-1c8a-08da9040ef46
x-ms-traffictypediagnostic: SJ0PR10MB5598:EE_
x-ms-exchange-senderadcheck: 1
x-ms-exchange-antispam-relay: 0
x-microsoft-antispam: BCL:0;
x-microsoft-antispam-message-info: 
 Y4WMskcYBciY1JZuSDldNxFr6QHygbEFWTM59AEA+afWBxSzzEXflbRbZoCq612aFJaIoD2YSC/dqCiTv2RGHamI36HT7KnElyXSiv3r3U9J7zpe8NONY90IrVbp3JdkOQpr41oOIP52WQ3RZQulJXs/MXX7CqHbmxMh7R9RbM954yCEhdu8Ak37OggT52hqnjRT9KrwOHocnNsfSUx8G6voOeguNHyesrSjSIcZJ6PtxraUeAD+/csQ8aQjD088krmYum6sdPz8myoTT321CgltcXIA1YyxLA2akqd7LwORuJVpqRdRTp7YNdc8TF0RT9ZNS/as8j+mJfURmMdQ0DoZdfV3UuUnSZ48XUx+sLsl+Oe+iShcKmXyJNWNfvayax6suD/yVIMMYmLWLIGTM+0WG/BruLcNwJbJk9SmN5GCk/i3Y/P7Do7w+hL2E9vxihE9e0CV7KdKs+iTsR55ikiysD1sh90/RmPmbuv+CFfeEq6eRYuFkaOCKTSTv9RG8GHQxMh4K9j0j/tFKlOXLBK6IEB0DE4/OYf21XxQ4gG7hoLpXZmAf2jkevp1vQpF8FotVObkxC0nYAq7OhdAVhWX3A61CaqjcvSPi/VrpOBuJIhwDAhX/w4MqHiEeOALlbWh/dROtm5qDB6e6254UMhOVGt3rY5LxNUdJtQRt47WivlGXHCj1QYpQvqxjb9KUHjKlRaWz5CFQ6HwVTyfDLv9gEWroc8idU5s24z3exPwatvFrYn8JI7HDQj12d47N0DN2/t3kCZPlZu7eKYXVA==
x-forefront-antispam-report: 
 CIP:255.255.255.255;CTRY:;LANG:en;SCL:1;SRV:;IPV:NLI;SFV:NSPM;H:SN6PR10MB3022.namprd10.prod.outlook.com;PTR:;CAT:NONE;SFS:(13230016)(346002)(39860400002)(396003)(136003)(376002)(366004)(64756008)(66556008)(41300700001)(91956017)(76116006)(66476007)(66446008)(66946007)(71200400001)(54906003)(110136005)(36756003)(316002)(6512007)(44832011)(8676002)(122000001)(2906002)(6486002)(4326008)(186003)(8936002)(5660300002)(478600001)(38070700005)(38100700002)(2616005)(86362001)(26005)(1076003)(6666004)(6506007)(83380400001);DIR:OUT;SFP:1101;
x-ms-exchange-antispam-messagedata-chunkcount: 1
x-ms-exchange-antispam-messagedata-0: =?iso-8859-1?q?CDXm9qQfkqqeVl3ZW7sUOjy?=
	=?iso-8859-1?q?lGX93t2roRWpDm204YJLjwnHUA5r1+roOi2XyxuGYnsUYC8JeE/BGrCjp9X7?=
	=?iso-8859-1?q?O+QxJpoiu6iAr4NJNAc6bbtGkY4W0OcleWrWMZ2mlPOnq4JAJXQU5phXXP6D?=
	=?iso-8859-1?q?ZN5hsH5WCrFXLaL7d1s0Kr2n5mnArOEh70HtAwvzNZyy9nrVF2VjrRx/cpkL?=
	=?iso-8859-1?q?WDghZgBENu3sdibC2Uxv+1ww4WmmkvjUGJCf7jh3+3//m5LoWXsEhKXhDb+C?=
	=?iso-8859-1?q?YhxRY6friIV+dP4ayJW0nsLbA5hPkeWNf5a7XQq2aH/iZQjxQqJ8RBbmnbB/?=
	=?iso-8859-1?q?mH53TjjUbAdg3yqrkth9br/sjRNbKw2pNsxqvnwxJvSd2JHoyuvj03xyWcFl?=
	=?iso-8859-1?q?TArQ9su/DB0j+rPjy4WDyCO7PLWft0qtsWn1lHWuTuAHfPQfqqOA2/Z4ZZgm?=
	=?iso-8859-1?q?9oGGLRMK/0Ukw0T8DQuOLunD54qacpIedDc24Nh/ANqhg4V0q+Zam+Hkxd2o?=
	=?iso-8859-1?q?FZfJ7qNMYKC+yAD5SyB0YfZ8Cxr/u4USSFINsWN1OXTQMXCe1H/CXIW68kQz?=
	=?iso-8859-1?q?mwKHuJg8bUo43sJJ4ymdUdEVDG9UbmSMzs7PM8yQwdwmGS/ISzHEWuNn5VaY?=
	=?iso-8859-1?q?UFjVpqo0kG8O9XoqKmO1dSQUKyrQYotwguVMOj35Cq7TEZCBsIzw0lSVKO8m?=
	=?iso-8859-1?q?4pdU6cn5ge1WMOZwUXWk89RgByYDF2UbK7qUSDA8P8M47eFf+F/tbv9UCAsv?=
	=?iso-8859-1?q?uPs8mIJZxkS5LsAEXBd7I62Tsl72Lkz2O+Uzpwskv4cfgQkfvPG/IMeqUnh+?=
	=?iso-8859-1?q?X0j8ju1ge6gMiCRfb/+RRjPIqEMZXwbek/HSh0gwc3erF0VsMgLt7yGKmhMP?=
	=?iso-8859-1?q?SviuNP36iwkJk9plkxocWfSfIsrQowxZil5CwkD0+USYrIBEPX0dyJe0HmnK?=
	=?iso-8859-1?q?r5L8L1ziw8CCnxHzJzFrGAgHzgryZpRhqaZfcyNF02GYFxnKPxX/2s49tqNP?=
	=?iso-8859-1?q?0tN9RsK9Enp9EPiYfFL+dkT7OEfSxyu38JN1tNhMgfB4Gs5mQeVvGGcdqqsd?=
	=?iso-8859-1?q?ubEre64yQmcfI036aJSfmxDzSrYsbRxVE81276+Dm6moz4ULB0J2j1C3uqXE?=
	=?iso-8859-1?q?VERVzXuJmPbtCiYW0TV+o1UVD5NP+d9wwfOQmLjsAXDeq39cMcJWOD1WwifV?=
	=?iso-8859-1?q?p+sDr8unWPgq+zkAe9mPRdF6YOaCXmqGuJZ9yBrfiycJbLv7z5X28r71Xxsg?=
	=?iso-8859-1?q?5vD8MVLVhla+RhT/oHBVdcW5F0N5Q5PeE+yYZltrFKN3b3bEGirUqqip0QNG?=
	=?iso-8859-1?q?hZugn373Tnk1kAqcxUSBALgqJE8V/pkfEw9znuQbj4Rsd+6pza4YTineU2jT?=
	=?iso-8859-1?q?yaO1HPiShXBtrgX9gN00S858zHaFW0aIdfVJgQfEzhIHf11J4R9w4mDA8zIo?=
	=?iso-8859-1?q?pnB5cSs9jC1sxXIrs1Jkfv3qN321d1v0Qq+47k/HUPAHKlfo7p6D/xp1Up8p?=
	=?iso-8859-1?q?Ilw+4NiG/DrXhAvtYE1r1kw4p4YbAv/6Zz5wk5qxuEIr471WO3BxOFX3oBXU?=
	=?iso-8859-1?q?FlxD+8IbQ2rjRftN9kHlZqPuALzqx7e1NftUiVlhgKJluyy4CyyVH70ckLD3?=
	=?iso-8859-1?q?A6KQxzOxqcKPcf62f8ctLFFSxkgji34UaA6az4A=3D=3D?=
MIME-Version: 1.0
X-OriginatorOrg: oracle.com
X-MS-Exchange-CrossTenant-AuthAs: Internal
X-MS-Exchange-CrossTenant-AuthSource: SN6PR10MB3022.namprd10.prod.outlook.com
X-MS-Exchange-CrossTenant-Network-Message-Id: 
 af2b36f8-81cf-412b-1c8a-08da9040ef46
X-MS-Exchange-CrossTenant-originalarrivaltime: 06 Sep 2022 19:48:56.1156
 (UTC)
X-MS-Exchange-CrossTenant-fromentityheader: Hosted
X-MS-Exchange-CrossTenant-id: 4e2c6054-71cb-48f1-bd6c-3a9705aca71b
X-MS-Exchange-CrossTenant-mailboxtype: HOSTED
X-MS-Exchange-CrossTenant-userprincipalname: 
 wHlZeGzUkLdCif4qdWKDBV1X0DO7HYwXs8yAgRyRcT3hjwZBbyBb1DtatiTjwXkEjEn68s0V2dUSKSmztb93kQ==
X-MS-Exchange-Transport-CrossTenantHeadersStamped: SJ0PR10MB5598
X-Proofpoint-Virus-Version: vendor=baseguard
 engine=ICAP:2.0.205,Aquarius:18.0.895,Hydra:6.0.528,FMLib:17.11.122.1
 definitions=2022-09-06_09,2022-09-06_02,2022-06-22_01
X-Proofpoint-Spam-Details: rule=notspam policy=default score=0 bulkscore=0
 mlxlogscore=999
 phishscore=0 mlxscore=0 spamscore=0 malwarescore=0 suspectscore=0
 adultscore=0 classifier=spam adjust=0 reason=mlx scancount=1
 engine=8.12.0-2207270000 definitions=main-2209060091
X-Proofpoint-ORIG-GUID: lf6zPRcBhB6ch09qKKo7GMOGApqB_AZD
X-Proofpoint-GUID: lf6zPRcBhB6ch09qKKo7GMOGApqB_AZD
ARC-Message-Signature: i=2; a=rsa-sha256; c=relaxed/relaxed;
 d=hostedemail.com;
	s=arc-20220608; t=1662493790;
	h=from:from:sender:reply-to:subject:subject:date:date:
	 message-id:message-id:to:to:cc:cc:mime-version:mime-version:
	 content-type:content-type:
	 content-transfer-encoding:content-transfer-encoding:
	 in-reply-to:in-reply-to:references:references:dkim-signature;
	bh=GJ+7dVHdLllKMNXTH8C4IlcGRwqY5aNlvrMEbXEver4=;
	b=jfPMx4gRPe9HmLG9j1BSeU4f7MyHXfN5bCcMeLRTmscCGGEpZWRaqikuVWP7Bu923+n1zC
	vzTy+LziX3sFlrdh2hSVBWN2/IZSO4kohowPRwQsjUbiswrdvquUEAj5sVMaEsjieAigw7
	yEgIskQ1fZD+T5AZOvXtoBylApecDDA=
ARC-Authentication-Results: i=2;
	imf20.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=ObTpfPtU;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=HOJ2NkfY;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf20.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
ARC-Seal: i=2; s=arc-20220608; d=hostedemail.com; t=1662493790; a=rsa-sha256;
	cv=pass;
	b=jBd28kfuNzK8352bb5/qaEddkCBbLHoDeE2rLrk3J8BeFgxZpG2jJHeaBiYbRDvbuSupGp
	lxeyq1qjZNsGvtiogMDfNfr08HcHdVsln7RhL/JjrHMCNIFs2g0bf4vyorSV8mXPRXSDJ+
	Kw7hQSLFVulxaUzf7NjjkJ4/RhHmYic=
X-Stat-Signature: 116k3cuyemfgxac6eusgite3wrnqz9q3
X-Rspamd-Queue-Id: 4A6C11C0091
X-Rspam-User: 
Authentication-Results: imf20.hostedemail.com;
	dkim=pass header.d=oracle.com header.s=corp-2022-7-12 header.b=ObTpfPtU;
	dkim=pass header.d=oracle.onmicrosoft.com
 header.s=selector2-oracle-onmicrosoft-com header.b=HOJ2NkfY;
	arc=pass ("microsoft.com:s=arcselector9901:i=1");
	spf=pass (imf20.hostedemail.com: domain of liam.howlett@oracle.com designates
 205.220.165.32 as permitted sender) smtp.mailfrom=liam.howlett@oracle.com;
	dmarc=pass (policy=none) header.from=oracle.com
X-Rspamd-Server: rspam03
X-HE-Tag: 1662493790-662298
X-Bogosity: Ham, tests=bogofilter, spamicity=0.000000, version=1.2.4
Sender: owner-linux-mm@kvack.org
Precedence: bulk
X-Loop: owner-majordomo@kvack.org
List-ID: <linux-mm.kvack.org>

From: "Matthew Wilcox (Oracle)" <willy@infradead.org>

Use the VMA iterator instead.

Signed-off-by: Matthew Wilcox (Oracle) <willy@infradead.org>
Signed-off-by: Liam R. Howlett <Liam.Howlett@Oracle.com>
Reviewed-by: Davidlohr Bueso <dave@stgolabs.net>
---
 arch/um/kernel/tlb.c | 14 ++++++--------
 1 file changed, 6 insertions(+), 8 deletions(-)

diff --git a/arch/um/kernel/tlb.c b/arch/um/kernel/tlb.c
index bc38f79ca3a3..ad449173a1a1 100644
--- a/arch/um/kernel/tlb.c
+++ b/arch/um/kernel/tlb.c
@@ -584,21 +584,19 @@ void flush_tlb_mm_range(struct mm_struct *mm, unsigned long start,
 
 void flush_tlb_mm(struct mm_struct *mm)
 {
-	struct vm_area_struct *vma = mm->mmap;
+	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
-	while (vma != NULL) {
+	for_each_vma(vmi, vma)
 		fix_range(mm, vma->vm_start, vma->vm_end, 0);
-		vma = vma->vm_next;
-	}
 }
 
 void force_flush_all(void)
 {
 	struct mm_struct *mm = current->mm;
-	struct vm_area_struct *vma = mm->mmap;
+	struct vm_area_struct *vma;
+	VMA_ITERATOR(vmi, mm, 0);
 
-	while (vma != NULL) {
+	for_each_vma(vmi, vma)
 		fix_range(mm, vma->vm_start, vma->vm_end, 1);
-		vma = vma->vm_next;
-	}
 }

